daemon/state.go

```diff
@@ -149,9 +149,18 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) {
 		ep.Mutex.RUnlock()
 
 		go func(ep *endpoint.Endpoint, epRegenerated chan<- bool) {
-			ep.Mutex.Lock()
-
+			ep.Mutex.RLock()
 			scopedLog := log.WithField(logfields.EndpointID, ep.ID)
+			// Filter the restored labels with the new daemon's filter
+			l, _ := labels.FilterLabels(ep.OpLabels.IdentityLabels())
+			ep.Mutex.RUnlock()
+
+			identity, _, err := identityPkg.AllocateIdentity(l)
```

Comment:
Where is the actual change in this commit? This was already in the background.

Reply:
The endpoint lock was being held while doing a new identity allocation, which is itself a blocking function. This would block `cilium endpoint list` as we couldn't retrieve the list of endpoints since they would all be blocked while waiting for a new identity..